app.js

```diff
@@ -24,82 +24,93 @@ class Results extends React.Component {
     } else {
       newList = currentList;
     }
-    this.setState({
-      results: newList
-    });
+
+    // this.setState({
+    //   results: newList
+    // });
   }
 
   render() {
     return (
       <div>
-        <input type="text" className="input" onChange={this.handleChange} placeholder="Search..." />
-          <ul>
-            {this.state.results.map(item => (
-              <li key={item}>
-                {item} &nbsp;
-              </li>
-            ))}
-          </ul>
+        <input
+          type="text"
+          className="input"
+          onChange={this.handleChange}
+          placeholder="Search..."
+        />
+        <ul>
+          {this.props.results.map(item => (
+            <li key={item}>{item} &nbsp;</li>
+          ))}
+        </ul>
       </div>
     );
   }
 }
 
 class App extends React.Component {
-    constructor(props) {
-	super(props);	
-	this.state = {
-	    isFetching: true,
-	    results: ["Afghanistan",
-		      "Åland Islands",
-		      "Albania",
-		      "Algeria"]
-	};
-    }
+  constructor(props) {
+    super(props);
+    this.state = {
+      isFetching: true,
+      results: ["Afghanistan", "Åland Islands", "Albania", "Algeria"]
+    };
+  }
 
-    componentDidMount() {
-	console.log("in component did mount")
+  componentDidMount() {
+    const doc = new GoogleSpreadsheet(
+      "194rof1NCeXCIGIaPLQu61mj23-YeIFyJA4hQc8q7UQE",
+      null,
+      { gzip: false }
+    );
+
+    doc.getInfo((err, info) => {
+      if (err) {
+        console.log(err);
+        return;
+      }
+      const sheet = info.worksheets[0];
+
+      sheet.getRows({ offset: 1, limit: 11 }, (err, response) => {
+        if (err) {
+          console.log(err);
+          return;
+        }
 
-	let doc = new GoogleSpreadsheet('194rof1NCeXCIGIaPLQu61mj23-YeIFyJA4hQc8q7UQE', null, {gzip: false});
-	let sheet;
-	let names;
-	let rows = [];
-	doc.getInfo(function(err, info) {
-	    if( err ) {
-		console.log(err);
-	    }
-	    console.log('Loaded doc: '+info.title+' by '+info.author.email);
-	    sheet = info.worksheets[0];
-	    console.log('sheet 1: '+sheet.title+' '+sheet.rowCount+' x '+sheet.colCount);
-	    sheet.getRows({offset: 1, limit: 11}, function(err, rs) {
-		if( err ) {
-		    console.log(err);
-		}
-		rows=rs;
-		console.log("example row: " + rows[10].comedyclubfestival);
-		console.log("getRows rows: " + rows);
-	    });
-	    console.log("getInfo rows: " + rows);
+        console.log(response);
+
+        // do your actually data mapping here,
+        const results = response.map(row => {
+          return row.comedyclubfestival;
+        });
+
+        this.setState(
```

Comment:
arrow functions () => {} don't bind 'this' so you can still access this.setState from this level of nesting